libminifi/include/core/ProcessContext.h

```diff
@@ -193,6 +216,61 @@ class ProcessContext : public controller::ControllerServiceLookup, public core::
     return controller_service_provider_->getControllerServiceName(identifier);
   }
 
+  static constexpr char const* DefaultStateManagerProviderName = "defaultstatemanagerprovider";
```

Comment:
> default- ok
> state- stateless is generally better, but ok
This is a PR about creating a centralized state storage mechanism for processors that require state, instead of all of them implementing individual state files. "Stateless is generally better" is a mind-bogglingly meaningless comment here.
> manager- meaningless. Only suggests doing something, so it should probably be a verb. Having a noun in place of a verb usually means trying to use a class where a function would be better.
StateManager comes from NiFi, and it is a perfectly meaningful name: it is an object that helps manage the state of a single CoreComponent. It cannot be a function as it contains some caching and validation logic that requires it in itself to be stateful.
> provider- again, a class instead of a function. This has a meaning though.
CoreComponentStateManagerProvider needs to exist as interface, which is used by the ClassLoader and we access different implementations of it through that interface.
> name- ok
> 
> I'm not saying that your code is bad, just that this is a red flag that suggests a design mistake that may or may not be in your changes.
I'm not saying your comment is useless, but it could be more useful if you tried to interpret the name in the context of the PR, instead of in itself.

Reply:
Context is always important when reviewing code. Most code has a history even if we don't like it, so it may be useful to appreciate where it comes from and how we intend to get it where we want it to go.

Reply:
Sorry for coming through as offensive. I lack a lot of contextual knowledge and at the time of writing the original comment, I knew even less.

I maintain my standpoint that there is most likely too much abstraction but now I understand that changing this would require redesign of significant parts of the project and it is harmful to do so without serious thought.